google/cloud/storage/internal/storage_round_robin.cc

```diff
@@ -262,6 +281,32 @@ std::shared_ptr<StorageStub> StorageRoundRobin::Child() {
   return child;
 }
 
+void StorageRoundRobin::Refresh(
+    std::size_t index,
+    std::weak_ptr<google::cloud::internal::CompletionQueueImpl> wcq) {
+  auto cq = wcq.lock();
+  if (!cq) return;
+  auto deadline = std::chrono::system_clock::now() + kRefreshPeriod;
+  // An invalid index, stop the loop.  There is no need to lock the mutex, as
+  // the channels do not change after the class is initialized.
```

Comment:
Q.  How is an invalid index injected in the first place?  If `channels_` doesn't change after initialization, couldn't we have caught this earlier?

Reply:
This is just defensive.  It really cannot (or should not) happen, but if it does then I would rather not crash and just stop using the invalid index.

Reply:
OK, thanks, although I've never been a big fan of that philosophy.